src/ConnectedMode.UnitTests/Migration/XmlFileCleanerTests.cs

```diff
@@ -157,7 +158,44 @@ public void Clean_RulesetRefsExist_DifferentProjectKey_SettingsAreNotRemoved()
             var testSubject = CreateTestSubject();
 
             var actual = testSubject.Clean(input, settings, CancellationToken.None);
-            actual.Should().Be(XmlFileCleaner.Unchanged);
+            CheckIsUnchanged(actual);
+        }
+
+        [TestMethod]
+        public void Clean_None_CorrectProjectKey_SettingsAreRemoved()
```

Comment:
Two new tests, in the same pattern as the existing ones i.e.
1) the first checks the file is cleaned as expected if you use the correct project key, and
2) the second checks nothing is removed if the project key doesn't match.